sema/commons/j2/j2_functions.py

```
@@ -82,6 +82,36 @@
+    if isinstance(content, date):
+        year = content.year  # extract year from date
+    else:
+        content = str(content).strip()  # other types handled via trimmed string
+        year = int(content)  # convert to int
+    # we should be sure of int now
+    # see https://www.datypic.com/sc/xsd11/t-xsd_gYear.html for examples of correct values
+    content = f"{"-" if year < 0 else ""}{abs(year):04d}"
+    return xsd_value(content, quote, "xsd:gYear")
+
+
+def xsd_format_gmonthyear(content, quote, suffix):
```

Ruff (0.8.2)

Missing return type annotation for public function `xsd_format_gmonthyear`

(ANN201)